salt/knox/init.sls

```diff
@@ -9,7 +10,12 @@
 {% set hive_host = salt['pnda.get_hosts_by_hadoop_role']('HIVE', 'HIVE_SERVER')[0] %}
 {% set webhdfs_host = salt['pnda.get_hosts_by_hadoop_node']('MGR01')[0] %}
 {% set hbase_rest_host = salt['pnda.get_hosts_by_hadoop_node']('MGR01')[0] %}
-{% set yarn_rm_host = salt['pnda.get_hosts_by_hadoop_role']('YARN', 'RESOURCEMANAGER')[0] %}
+{% set yarn_rm_hosts = salt['pnda.get_hosts_by_hadoop_role']('YARN', 'RESOURCEMANAGER') %}
+{% if yarn_rm_hosts is not none and yarn_rm_hosts|length > 1 %}
```

Comment:
I think the code would be clearer if we either -
 - Just use the predicate (yarn_rm_hosts is not none and yarn_rm_hosts|length > 1) at the point where we make the decision to include the HA XML
or;
- Rename this additional boolean to yarn_ha_enabled and set it to (yarn_rm_hosts is not none and yarn_rm_hosts|length > 1) rather than having a general name and all the additional if/then/else syntax